protocol-units/bridge/chains/movement/src/lib.rs

```diff
@@ -458,13 +461,18 @@ impl BridgeContractCounterparty for MovementClient {
 		amount: Amount,
 	) -> BridgeContractCounterpartyResult<()> {
 
+		let amount_value = match amount.0 {
+			AssetType::Moveth(value) => value,
+			_ => return Err(BridgeContractCounterpartyError::SerializationError),
+		};
+
 		let args = vec![
-			utils::serialize_vec(&initiator.0)?,
-			utils::serialize_vec(&bridge_transfer_id.0[..])?,
-			utils::serialize_vec(&hash_lock.0[..])?,
-			utils::serialize_u64(&time_lock.0)?,
-			utils::serialize_vec(&recipient.0.0.to_vec())?,
-			utils::serialize_u64(&amount.moveth())?,
+			bcs::to_bytes(&initiator.0).map_err(|_| BridgeContractCounterpartyError::SerializationError)?,
```

Comment:
I recall we created, or discussed using a util fn that wraps `bcs::to_bytes` and returns a `Result<T,  BridgeContractCounterpartyError::SerializationError>` reducing the repeating `.map_err` calls.

Reply:
Yeah those utility functions were was causing a `CallError` but I just figured out why and fixed it. Good to go.